https://git.kernel.org/pub/scm/libs/ell/ell.git/commit/?id=ce7fcfe194f0abcb8f419f83276b16a4ab274032

From: "Milan P. StaniÄ‡" <mps@arvanta.net>
Date: Mon, 6 Jun 2022 22:05:39 +0200
Subject: build: fix time.h related breakage on musl

missing time.h for struct timeval usage
forward declaration of struct timeval in time-private.h
--- a/ell/dhcp-transport.c
+++ b/ell/dhcp-transport.c
@@ -40,6 +40,7 @@
 #include <linux/filter.h>
 #include <net/if_arp.h>
 #include <errno.h>
+#include <sys/time.h>
 
 #include "io.h"
 #include "util.h"
--- a/ell/dhcp6-transport.c
+++ b/ell/dhcp6-transport.c
@@ -35,6 +35,7 @@
 #include <net/if.h>
 #include <unistd.h>
 #include <errno.h>
+#include <sys/time.h>
 
 #include "private.h"
 #include "missing.h"
--- a/ell/icmp6.c
+++ b/ell/icmp6.c
@@ -36,6 +36,7 @@
 #include <net/if.h>
 #include <unistd.h>
 #include <errno.h>
+#include <sys/time.h>
 
 #include "private.h"
 #include "useful.h"
--- a/ell/time-private.h
+++ b/ell/time-private.h
@@ -19,7 +19,7 @@
  *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  *
  */
-
+struct timeval;
 uint64_t _time_pick_interval_secs(uint32_t min_secs, uint32_t max_secs);
 uint64_t _time_fuzz_msecs(uint64_t ms);
 uint64_t _time_fuzz_secs(uint32_t secs, uint32_t max_offset);
--- a/ell/time.c
+++ b/ell/time.c
@@ -26,6 +26,7 @@
 
 #define _GNU_SOURCE
 #include <time.h>
+#include <sys/time.h>
 
 #include "time.h"
 #include "time-private.h"
cgit 
